src/Azure.Functions.Sdk/Tasks/Extensions/ResolveExtensionPackages.cs

```diff
@@ -0,0 +1,171 @@
+    public ITaskItem[] ExtensionPackages { get; private set; } = [];
+
+    public void Cancel()
+    {
+        _cts.Cancel();
+    }
+
+    public void Dispose()
+    {
+        _cts.Dispose();
+    }
+
+    public override bool Execute()
+    {
+        if (!TryGetLockFile(out LockFile? lockFile))
+        {
+            return false;
+        }
+
+        List<ITaskItem> extensionPackages = [];
+        FallbackPackagePathResolver resolver = lockFile.GetPathResolver();
+        foreach (LockFileTarget target in lockFile.Targets)
+        {
+            if (!string.IsNullOrEmpty(target.RuntimeIdentifier))
+            {
+                // Scanning specific RID's will be redundant as packages can only condition on TFM.
+                continue;
+            }
+
+            if (_cts.IsCancellationRequested)
+            {
+                return false;
+            }
+
+            foreach (ITaskItem ext in GetExtensionPackages(target, resolver))
+            {
+                extensionPackages.Add(ext);
+            }
+        }
+
+        if (_cts.IsCancellationRequested)
+        {
+            return false;
+        }
+
+        ExtensionPackages = [.. extensionPackages];
+        return !Log.HasLoggedErrors;
+    }
+
+    private static bool ShouldScanLibrary(LockFileTargetLibrary library)
+    {
+        return library.Type == LibraryType.Package
+            && library.Name is not null
+            && !FunctionsAssemblyScanner.IsExcludedPackage(library.Name);
+    }
+
+    private bool TryGetLockFile([NotNullWhen(true)] out LockFile? lockFile)
+    {
+        if (_cts.IsCancellationRequested)
+        {
+            lockFile = null;
+            return false;
+        }
+
+        if (!_fileSystem.File.Exists(ProjectAssetsFile))
+        {
+            Log.LogError(Strings.MissingProjectAssetsFile, ProjectAssetsFile);
+            lockFile = null;
+            return false;
+        }
+
+        IFileInfo info = _fileSystem.FileInfo.New(ProjectAssetsFile);
+        using FileSystemStream stream = info.OpenRead();
+        LockFileFormat format = new();
+        lockFile = format.Read(stream, new MSBuildNugetLogger(Log), ProjectAssetsFile);
+        return true;
+    }
+
+    private IEnumerable<ITaskItem> GetExtensionPackages(LockFileTarget target, FallbackPackagePathResolver resolver)
+    {
+        foreach (LockFileTargetLibrary library in target.Libraries)
+        {
+            if (_cts.IsCancellationRequested)
+            {
+                yield break;
+            }
+
+            if (!ShouldScanLibrary(library))
+            {
+                continue;
+            }
+
+            string packagePath = resolver.GetPackageDirectory(library.Name, library.Version);
+            foreach (LockFileItem assembly in library.RuntimeAssemblies)
+            {
+                string path = _fileSystem.Path.Combine(packagePath, assembly.Path);
+                if (TryGetExtensionReference(path, library, out ITaskItem? ext))
```

Comment:
Do we account for duplicates - is that possible? What happens if there is a dupe

Reply:
A duplicate extension reference? A later task will be checking for duplicates.